File: src/main/java/com/github/graqr/threshr/model/redsky/products/Price.java
package com.github.graqr.threshr.model.redsky.products;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;
import jakarta.annotation.Nullable;

@Serdeable
public record Price(
        @JsonProperty("formatted_current_price") String formattedCurrentPrice,
        @JsonProperty("formatted_current_price_type") String formattedCurrentPriceType,
        @JsonProperty("reg_retail") Double regRetail,
        @JsonProperty("formatted_unit_price") String formattedUnitPrice,
        @JsonProperty("formatted_unit_price_suffix") String formattedUnitPriceSuffix,
        @JsonProperty("location_id") Long locationId,
        @Nullable @JsonProperty("current_retail") Double currentRetail,
        @Nullable @JsonProperty("external_system_id") String externalSystemId,
        @Nullable @JsonProperty("is_current_price_range") Boolean isCurrentPriceRange) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/Price.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable
import jakarta.annotation.Nullable

@Serdeable
data class Price(
    @param:JsonProperty("formatted_current_price") val formattedCurrentPrice: String,
    @param:JsonProperty("formatted_current_price_type") val formattedCurrentPriceType: String,
    @param:JsonProperty("reg_retail") val regRetail: Double,
    @param:JsonProperty("formatted_unit_price") val formattedUnitPrice: String,
    @param:JsonProperty("formatted_unit_price_suffix") val formattedUnitPriceSuffix: String,
    @param:JsonProperty("location_id") val locationId: Long,
    @param:Nullable @param:JsonProperty("current_retail") val currentRetail: Double,
    @param:Nullable @param:JsonProperty("external_system_id") val externalSystemId: String,
    @param:Nullable @param:JsonProperty("is_current_price_range") val isCurrentPriceRange: Boolean
) {}
